_posts/2020-06-20-Basic-Querying.md

```diff
@@ -0,0 +1,302 @@
+
+It is not clear how these traditions relate to each other or which is of greater antiquity. We are, however, presented with the intriguing situation in which two neighbouring kingdoms, Cíannachta Glinne Geimen and Dál nAraide, have origin legends in which satirists (one female and one sometimes female) play an important supporting role. This could, of course, be mere coincidence.
+
+In any case, this brief investigation shows how IrishGen can be queried for details other than names and genealogical relationship and thus used to compare details of narratives preserved across the genealogical collections.
+
+## Identifying Individuals at the Burning of the Kings
+Returning to the Cíannachta Glinne Geimen, after the eight (or nine) sons of Findchán, their history, as preserved in the genealogies, consists of string pedigrees unaccompanied by further comment. The individuals who appear in these pedigrees, however, can be located in other sources, such as the Irish annals. In this final section, we are going to look at one annal entry mentioning a king of Cíannachta Glinne Geimen and show how IrishGen can be used to identify the other characters in the entry.
+
+The entries in the _[Annals of Ulster](https://www.vanhamel.nl/codecs/Annals_of_Ulster)_ for 681 include the following ([681.1](https://celt.ucc.ie//published/T100001A/text251.html)):
+
+> Combustio regum i n-Dun Ceithirnn, .i. Dungal mc. Scannail, rex Cruithne, & Cenn Faelad, rex Cianachtae, .i. m. Suibni, in initio estatis la Mael Duin m. Maele Fitrich.
+> > The burning of the kings in Dún Ceithirn, i.e. Dúngal son of Scannal, king of the Cruithin, and Cenn Faelad, i.e. son of Suibne, king of Ciannachta by Mael Dúin son of Mael Fithrich, at the beginning of summer.
+
+Cenn Faelad, son of Suibne, appears in one of the royal pedigrees of the Cíannachta Glinne Geimen in **L** and **R** (Text 4). Who, then, are Dúngal, son of Scandal, alongside whom Cenn Faelad was burned, and Mael Dúin, son of Máel Fithrich, who is said to have been responsible? There are, of course, numerous works of secondary literature that could be consulted for context on this entry. For the purposes of this blogpost, however, we will search for the individuals named in IrishGen.
+
+The following SPARQL query searches IrishGen for every instance of someone called Dúnlang who is the child of someone called Scandal. Names usually appear in the genealogies in the genitive, we will begin by searching just the genitive forms of the two names. As with our query on _cáinte_, an effort is made to anticipate possible variant spellings.
+
+```
+PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+PREFIX rel: <http://purl.org/vocab/relationship/>
+PREFIX irishRel: <http://example.com/earlyIrishRelationship.ttl#>
+SELECT ?s
+WHERE
+{
+?s rel:childOf ?o;
+   irishRel:genName ?s_name.
+?o irishRel:genName ?o_name.
+    FILTER ((regex(str(?s_name), "D(u|ú)nlaing(e|i)", "i")) && (regex(str(?o_name), "Sca(nn|n|nd)ail", "i")))
+}
+```
```

Comment:
You might be able to get away with using a blank node here:

```sparql
PREFIX foaf: <http://xmlns.com/foaf/0.1/>
PREFIX rel: <http://purl.org/vocab/relationship/>
PREFIX irishRel: <http://example.com/earlyIrishRelationship.ttl#>
SELECT ?s
WHERE
{
?s rel:childOf [
irishRel:genName ?o_name.
FILTER(regex(str(?o_name), "Sca(nn|n|nd)ail", "i")
   ];
   irishRel:genName ?s_name.

FILTER ((regex(str(?s_name), "D(u|ú)nlaing(e|i)", "i")) && ))
}
```

This is off the cuff so it may not work and it could make it harder to read but give it a shot to see if you can use it.  I believe that you could, this construction, move the second filter into the [] area but again, give it a try to see if it works and if you like the way it reads.

Reply:
Thanks. If it is harder to read, I am thinking perhaps not, in what is meant to be a set of easy examples? But if you want to dive in and add this as an alternative and explain its advantages, my offer of co-authorship still stands.

Reply:
I think we should leave your version then.  I do not want to steal your thunder :)